Core/source/core/pgp-key.ts

```diff
@@ -311,12 +316,13 @@ export class PgpKey {
     return { original: armored, normalized, keys: await Promise.all(keys.map(PgpKey.details)) };
   }
 
-  public static details = async (k: OpenPGP.key.Key): Promise<KeyDetails> => {
+  public static details = async (k: Key): Promise<KeyDetails> => {
     const keys = k.getKeys();
-    const algoInfo = k.primaryKey.getAlgorithmInfo();
-    const algo = { algorithm: algoInfo.algorithm, bits: algoInfo.bits, curve: (algoInfo as any).curve, algorithmId: openpgp.enums.publicKey[algoInfo.algorithm] };
-    const created = k.primaryKey.created.getTime() / 1000;
-    const exp = await k.getExpirationTime('encrypt');
+    const algoInfo = k.keyPacket.getAlgorithmInfo();
+    const algo = { algorithm: algoInfo.algorithm, bits: algoInfo.bits, curve: (algoInfo as any).curve,
```

Comment:
`(algoInfo as any)` could have been due to lousy types before. Maybe this can be improved now? Could be a separate PR, or here.

Reply:
fixed it